pkg/multitenant/cost_controlling_io.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+package multitenant
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/util/ioctx"
+)
+
+// DefaultBytesAllowedBeforeAccounting are how maybe bytes we will read.write before trying to wait for
+// RUs. The goal here is to avoid waiting in loops for the common case without allowing an
+// unbounded number of bytes read/written before accounting for them.
+const DefaultBytesAllowedBeforeAccounting = 128 << 20 // 128 MB
```

Comment:
does this deserve to be a setting ?

Reply:
or an envutil.WithDefault

128mb seems high to me -- I'd have expected 16 or less?